src/csharp/Microsoft.Spark/Sql/Catalog/Catalog.cs

```diff
@@ -0,0 +1,287 @@
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Spark.Interop.Ipc;
+
+namespace Microsoft.Spark.Sql.Catalog
+{
+    /// <summary>
+    /// Catalog interface for Spark. To access this, use SparkSession.Catalog.
+    /// </summary>
+    public sealed class Catalog : IJvmObjectReferenceProvider
+    {
+        private readonly JvmObjectReference _jvmObject;
+        internal Catalog(JvmObjectReference jvmObject)
+        {
+            _jvmObject = jvmObject;
+        }
+
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+
+        /// <summary>
+        /// Caches the specified table in-memory.
+        /// </summary>
+        /// <param name="tableName">Is either a qualified or unqualified name that designates a
+        /// table. If no database identifier is provided, it refers to a table in the current
+        /// database.</param>
+        public void CacheTable(string tableName) =>
+            _jvmObject.Invoke("cacheTable", tableName);
+
+        /// <summary>
+        /// Removes all cached tables from the in-memory cache.
+        /// </summary>
+        public void ClearCache() => _jvmObject.Invoke("clearCache");
+
+        /// <summary>
+        /// Creates a table from the given path and returns the corresponding DataFrame.
+        /// </summary>
+        /// <param name="tableName">The name of the table to create</param>
+        /// <param name="path">Path to use to create the table</param>
+        /// <returns>DataFrame</returns>
+        public DataFrame CreateTable(string tableName, string path) =>
+            new DataFrame((JvmObjectReference)_jvmObject.Invoke("createTable", tableName, path));
+
+        /// <summary>
+        /// Creates a table from the given path based on a data source and returns the corresponding
+        /// DataFrame.
+        /// </summary>
+        /// <param name="tableName">Is either a qualified or unqualified name that designates a
+        /// table. If no database identifier is provided, it refers to a table in the current
+        /// database.</param>
+        /// <param name="path">Path to use to create the table</param>
+        /// <param name="source"></param>
+        /// <returns>DataFrame</returns>
+        public DataFrame CreateTable(string tableName, string path, string source) =>
+            new DataFrame(
+                (JvmObjectReference)_jvmObject.Invoke("createTable", tableName, path, source));
+
+        /// <summary>
+        /// Returns the current default database in this session.
+        /// </summary>
+        /// <returns>string</returns>
+        public string CurrentDatabase() => (string)_jvmObject.Invoke("currentDatabase");
+
+        /// <summary>
+        /// Check if the database with the specified name exists.
+        /// </summary>
+        /// <param name="dbName">Name of the database to check</param>
+        /// <returns>bool</returns>
```

Comment:
Can this be more descriptive?

Reply:
I meant something like..
```
<returns>Returns true if ... false otherwise</returns>
```

Same comment for the other <returns>bool</returns> descriptions in the summary.